internal/handler/impl/utils.go

```diff
@@ -20,3 +24,27 @@ func getXRHID(ctx echo.Context) (*identity.XRHID, error) {
 		return xrhid, nil
 	}
 }
+
+func sendPendoTrackEvent(ctx echo.Context, client pendo.Pendo, event string) {
+	c := ctx.Request().Context()
+	logger := app_context.LogFromCtx(c)
+
+	xrhid, err := getXRHID(ctx)
+	if err != nil {
+		logger.Error(err.Error())
+		return
+	}
+
+	track := pendo.TrackRequest{
+		AccountID: xrhid.Identity.OrgID,
+		Type:      "track",
+		Event:     event,
+		VisitorID: header.GetPrincipal(xrhid),
+		Timestamp: time.Now().UTC().Unix(),
+	}
+
+	err = client.SendTrackEvent(c, &track)
+	if err != nil {
+		logger.Warn(err.Error())
+	}
+}
```

Comment:
idea: what if we add the function as a private method, the `client` is available as part of the attributes of the instance, and we would have it available:

for instance:

```suggestion
func (a *application) sendPendoTrackEvent(ctx echo.Context, event string) error {
	var (
		xrhid *identity.XRHID
		err   error
	)
	c := ctx.Request().Context()
	logger := app_context.LogFromCtx(c)

	if xrhid, err = getXRHID(ctx); err != nil {
		logger.Error(err.Error())
		return err
	}

	track := pendo.TrackRequest{
		AccountID: xrhid.Identity.OrgID,
		Type:      pendoTypeTrack,
		Event:     event,
		VisitorID: header.GetPrincipal(xrhid),
		Timestamp: time.Now().UTC().Unix(),
	}

	if err = a.pendo.SendTrackEvent(c, &track); err != nil {
		logger.Warn(err.Error())
		return err
	}

	logger.Debug("sendPendoTrackEvent was success",
		slog.String("Type", track.Type),
		slog.String("Event", track.Event))
	return nil
}
```

In the above case I would keep in a different file, for instance `internal_methods.go` (not using `private.go` as already we have `private_handler.go`, maybe I didn't choose the best name when the `private_handler.go` file was added).

Reply:
I don't see a big need to move it as method of app. 

If we would move it then it might be to have more specific filename. As both `utils` or  `internal`, "private" forms coincidental cohesion which is better to be avoided.   E.g. `pendo_helpers.go` might be better.